### PR TITLE
Fix Data Mismatch, Update Mapping Template and Return Response as JSON

### PR DESCRIPTION
**Pull request by EbrahimAli-RU**
Fixes # #188 

Resolves #
Updated the mapping template to ensure data consistency and modified the response to return JSON, addressing data mismatches for improved accuracy.

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [x] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

**Review by melon-ruet — APPROVED** [review]

File: commons/dto/DonationDTO.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export type DonationDTO = DTO & HasIdentifier & {
   geohash: string;
   donationDateTime: string;
   status: DonationStatus;
-  contactInfo: { name: string; phone: string };
+  contactNumber: string;
   patientName?: string;
```
Comment (sakibstark11) [inline commons/dto/DonationDTO.ts:23]: if this is optional, it should be a separate data type that extends the core data type.

File: core/application/bloodDonationWorkflow/Types.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export interface UpdateBloodDonationAttributes {
   bloodQuantity?: number;
   urgencyLevel?: UrgencyLevel;
   donationDateTime?: string;
-  contactInfo?: { name: string; phone: string };
+  contactNumber?: string;
```
Comment (sakibstark11) [inline core/application/bloodDonationWorkflow/Types.ts:31]: SAME AS BEFORE

File: commons/dto/DonationDTO.ts (hunk 1 above)
Comment (sakibstark11) [reply commons/dto/DonationDTO.ts:23]: the reason being, what if i want to force the type casting on a particular object? then I can't guarantee the validity of the object

**Review by mahinuzzaman — APPROVED** [review]

File: commons/dto/DonationDTO.ts (hunk 1 above)
Comment (EbrahimAli-RU) [reply commons/dto/DonationDTO.ts:25]: I see your point about separating types for clarity.

If I understand you correctly, you’re suggesting that if we keep required and optional fields in one type, we can't guarantee that the required fields will be present when we forcefully cast an object.

However, if we separate the required fields from the optional ones and then extend the required type, I believe we still cannot guarantee the validity of the object.

**Review by sakibstark11 — APPROVED** [review]

**Pull request merged**